File: sdk/fsharp/src/HiTechCloudSDK/Notifications.fs
namespace HiTechCloud.SDK

/// Notifications: manage notification preferences
type NotificationsResource(httpClient: HttpClient) =

    /// GET /api/notifications - List all notifications
    member this.List() =
        async {
            return! httpClient.getAsync("/api/notifications")
        }

    /// GET /api/notifications/new - Return only new portal notifications
    member this.ListNew() =
        async {
            return! httpClient.getAsync("/api/notifications/new")
        }

    /// PUT /api/notifications/{id}/ack - Mark notification as read
    member this.Acknowledge() =
        async {
            return! httpClient.putAsync("/api/notifications/{id}/ack")
        }

File: sdk/clojure/src/hitechcloud/sdk/notifications.clj
(ns hitechcloud.sdk.notifications
  (:require [hitechcloud.sdk.http-client :as http]))

;; Notifications: manage notification preferences

(defn list "GET /api/notifications - List all notifications" [client & [params]] (http-get client "/api/notifications" params))

(defn list_new "GET /api/notifications/new - Return only new portal notifications" [client & [params]] (http-get client "/api/notifications/new" params))

(defn acknowledge "PUT /api/notifications/{id}/ack - Mark notification as read" [client & [params]] (http-put client "/api/notifications/{id}/ack" params))

